tests/pyright/base.py

```diff
@@ -0,0 +1,18 @@
+import json
+import subprocess
+
+
+def run_pyright(filename):
+    result = subprocess.run(
+        ["pyright", "--outputjson", filename],
+        capture_output=True,
+        text=True,
+    )
+    assert result.stdout, result.stderr
+    output = json.loads(result.stdout)
+
+    def clean_row(data):
+        del data["file"]
+        return data
+
+    return [clean_row(row) for row in output["generalDiagnostics"]]
```

Comment:
Could you add a few comments to explain the various steps here ? Nothing fancy, just the obvious